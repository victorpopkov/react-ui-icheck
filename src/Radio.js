import PropTypes from 'prop-types';
import BaseInput from './BaseInput'; // eslint-disable-line sort-imports
import deprecated from './helpers/deprecated';

class Radio extends BaseInput {
  static propTypes = {
    ...BaseInput.propTypes,
    radioClass: deprecated(PropTypes.string, 'radioClassName'),
    radioClassName: PropTypes.string,
  };

  static defaultProps = {
    ...BaseInput.defaultProps,
    radioClass: null,
    radioClassName: 'iradio',
  };

  renderInput() {
    const { radioClass, radioClassName } = this.props;

    return super.renderInput({
      inputType: 'radio',
      radioClassName: radioClass || radioClassName,
    });
  }
}

export default Radio;
