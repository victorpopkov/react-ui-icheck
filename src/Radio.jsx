import PropTypes from 'prop-types';
import BaseInput from './BaseInput'; // eslint-disable-line sort-imports

class Radio extends BaseInput {
  renderInput() {
    const { radioClass, radioClassName } = this.props;

    return super.renderInput({
      inputType: 'radio',
      radioClassName: radioClass || radioClassName,
    });
  }
}

Radio.propTypes = {
  ...BaseInput.propTypes,
  radioClassName: PropTypes.string,
};

Radio.defaultProps = {
  ...BaseInput.defaultProps,
  radioClass: null,
  radioClassName: 'iradio',
};

export default Radio;
