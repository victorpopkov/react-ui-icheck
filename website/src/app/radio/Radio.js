import {
  Col,
  Container,
  Nav,
  NavItem,
  NavLink,
  Row,
  TabContent,
  TabPane,
} from 'reactstrap';
import { Radio, RadioGroup } from 'react-ui-icheck'; // eslint-disable-line import/no-unresolved
import React, { Component } from 'react';
import Prism from 'prismjs';
import classNames from 'classnames';
import { hot } from 'react-hot-loader';
import { Example, Jumbotron, Sidebar } from '../common'; // eslint-disable-line sort-imports
import RadioProperties from './RadioProperties';

@hot(module)
class RadioPage extends Component {
  constructor(props) {
    super(props);

    this.toggleActiveTab = this.toggleActiveTab.bind(this);

    this.state = {
      activeTab: 'examples',
      radioValue: '2',
    };
  }

  componentDidMount() {
    Prism.highlightAll();
  }

  toggleActiveTab(tab) {
    const { activeTab } = this.state;

    if (activeTab !== tab) {
      this.setState({ activeTab: tab });
    }
  }

  render() {
    const { activeTab, radioValue } = this.state;

    return (
      <main role="main">
        <Jumbotron />
        <Container>
          <Row>
            <Col md={9}>
              <h2 className="heading">Checkbox</h2>
              <Nav className="mt-4" tabs>
                <NavItem>
                  <NavLink
                    className={classNames({ active: activeTab === 'examples' })}
                    onClick={() => { this.toggleActiveTab('examples'); }}
                  >
                    Examples
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink
                    className={classNames({ active: activeTab === 'properties' })}
                    onClick={() => { this.toggleActiveTab('properties'); }}
                  >
                    Properties
                  </NavLink>
                </NavItem>
              </Nav>
              <TabContent
                activeTab={activeTab}
                className="mb-4"
              >
                <TabPane tabId="examples">
                  <Example
                    srcFunctionValue="(event, value) => this.setState({ radioValue: value })"
                    srcPrepend={`import 'icheck/skins/all.css';
import { Radio, RadioGroup } from 'react-ui-icheck';

this.state = {
  radioValue: '2',
};\n\n`}
                    title="Default"
                  >
                    <RadioGroup
                      name="radio"
                      radioWrapClassName="form-check"
                      radioWrapTag="div"
                      value={radioValue}
                      onChange={(event, value) => this.setState({ radioValue: value })}
                    >
                      <Radio
                        increaseArea="30%"
                        label="#radio-1"
                        radioClass="iradio_square-blue"
                        value="1"
                      />
                      <Radio
                        increaseArea="30%"
                        label="#radio-2"
                        labelTag="div"
                        labelTagClassName="d-inline"
                        radioClass="iradio_square-blue"
                        value="2"
                      />
                      <Radio
                        increaseArea="30%"
                        labelTag="div"
                        labelTagClassName="d-inline"
                        radioClass="iradio_square-blue"
                        value="3"
                        disabled
                      >
                        <span>#radio-3, </span>
                        <span className="label1">disabled</span>
                      </Radio>
                    </RadioGroup>
                  </Example>
                  <Example
                    srcFunctionValue="(event, value) => this.setState({ radioValue: value })"
                    srcPrepend={`import 'icheck/skins/all.css';
import { Radio, RadioGroup } from 'react-ui-icheck';

this.state = {
  radioValue: '2',
};\n\n`}
                    title="Inline"
                  >
                    <RadioGroup
                      className="d-flex align-items-center"
                      name="radio"
                      radioWrapClassName="form-check form-check-inline"
                      radioWrapTag="div"
                      value={radioValue}
                      onChange={(event, value) => this.setState({ radioValue: value })}
                    >
                      <Radio
                        label="#radio-1"
                        radioClass="iradio_square-blue"
                        value="1"
                      />
                      <Radio
                        label="#radio-2"
                        labelTag="div"
                        labelTagClassName="d-inline"
                        radioClass="iradio_square-blue"
                        value="2"
                      />
                      <Radio
                        labelTag="div"
                        labelTagClassName="d-inline"
                        radioClass="iradio_square-blue"
                        value="3"
                        disabled
                      >
                        <span>#radio-3, </span>
                        <span className="label1">disabled</span>
                      </Radio>
                    </RadioGroup>
                  </Example>
                </TabPane>
                <TabPane tabId="properties">
                  <RadioProperties />
                </TabPane>
              </TabContent>
            </Col>
            <Sidebar />
          </Row>
        </Container>
      </main>
    );
  }
}

export default RadioPage;
