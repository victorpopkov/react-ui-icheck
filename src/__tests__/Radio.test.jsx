import { mount, render } from 'enzyme';
import React from 'react';
import expect from 'expect';
import Radio from '../Radio'; // eslint-disable-line sort-imports

describe('<Radio /> component', () => {
  it('should have the default props values when mounted', () => {
    const wrapper = mount(<Radio />);
    expect(wrapper.props()).toMatchSnapshot();
  });

  describe('should render the corresponding HTML', () => {
    it('when only the default props are set', () => {
      const wrapper = render(<Radio />);
      expect(wrapper).toMatchSnapshot();
    });

    it('when the label is set as a prop string value', () => {
      const wrapper = render(<Radio label="label" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('when the label is set as children with labelTag prop set to <div />', () => {
      const wrapper = render(
        <Radio labelTag="div">
          <span>label</span>
        </Radio>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('should update the state', () => {
    const testRadio = <Radio label="label" />;

    it('when the checked prop has changed', () => {
      const spy = jest.spyOn(Radio.prototype, 'componentWillReceiveProps');
      const wrapper = mount(testRadio);

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.props().checked).toBe(false);
      expect(wrapper.state().checked).toBe(false);

      wrapper.setProps({ checked: true });

      expect(spy).toHaveBeenCalledTimes(1);
      expect(wrapper.props().checked).toBe(true);
      expect(wrapper.state().checked).toBe(true);
    });

    describe('when handling the <label />', () => {
      it('onFocus()', () => {
        const wrapper = mount(testRadio);

        expect(wrapper.state().hovered).toBe(false);
        wrapper.simulate('focus');
        expect(wrapper.state().hovered).toBe(true);
      });

      it('onBlur()', () => {
        const wrapper = mount(testRadio);

        wrapper.setState({ hovered: true });

        expect(wrapper.state().hovered).toBe(true);
        wrapper.simulate('blur');
        expect(wrapper.state().hovered).toBe(false);
      });

      it('onMouseOver()', () => {
        const wrapper = mount(testRadio);

        expect(wrapper.state().hovered).toBe(false);
        wrapper.simulate('mouseOver');
        expect(wrapper.state().hovered).toBe(true);
      });

      it('onMouseOut()', () => {
        const wrapper = mount(testRadio);

        wrapper.setState({ hovered: true });

        expect(wrapper.state().hovered).toBe(true);
        wrapper.simulate('mouseOut');
        expect(wrapper.state().hovered).toBe(false);
      });

      it('onTouchStart()', () => {
        const wrapper = mount(testRadio);

        expect(wrapper.state().hovered).toBe(false);
        wrapper.simulate('touchStart');
        expect(wrapper.state().hovered).toBe(true);
      });

      it('onTouchEnd()', () => {
        const wrapper = mount(testRadio);

        wrapper.setState({ hovered: true });

        expect(wrapper.state().hovered).toBe(true);
        wrapper.simulate('touchEnd');
        expect(wrapper.state().hovered).toBe(false);
      });
    });
  });
});
