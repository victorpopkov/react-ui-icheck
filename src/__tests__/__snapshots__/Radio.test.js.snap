// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> component should have the default props values when mounted 1`] = `
Object {
  "activeClass": "active",
  "aria": false,
  "checked": false,
  "checkedClass": "checked",
  "children": null,
  "className": null,
  "defaultChecked": false,
  "determinateClass": null,
  "disabled": false,
  "disabledClass": "disabled",
  "focusClass": "focus",
  "hoverClass": "hover",
  "id": null,
  "increaseArea": "",
  "indeterminate": true,
  "indeterminateClass": "indeterminate",
  "inheritClass": false,
  "inheritID": false,
  "insert": null,
  "isMobile": false,
  "label": null,
  "labelClassName": null,
  "labelHover": true,
  "labelHoverClass": "hover",
  "labelTag": "span",
  "labelTagClassName": null,
  "name": null,
  "onBlur": null,
  "onChange": null,
  "onFocus": null,
  "onMouseDown": null,
  "onMouseLeave": null,
  "onMouseUp": null,
  "onTouchEnd": null,
  "onTouchStart": null,
  "radioClass": "iradio",
  "type": null,
  "value": "",
}
`;

exports[`<Radio /> component should render the corresponding HTML when only the default props are set 1`] = `
<div
  class="iradio indeterminate"
>
  <input
    style="opacity:0;position:absolute"
    type="radio"
    value=""
  />
  <ins
    class="iCheck-helper"
    role="presentation"
    style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
  />
</div>
`;

exports[`<Radio /> component should render the corresponding HTML when the label is set as a prop string value 1`] = `
<label>
  <div
    class="iradio indeterminate"
  >
    <input
      style="opacity:0;position:absolute"
      type="radio"
      value=""
    />
    <ins
      class="iCheck-helper"
      role="presentation"
      style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
    />
  </div>
  <span>
    label
  </span>
</label>
`;

exports[`<Radio /> component should render the corresponding HTML when the label is set as children with labelTag prop set to <div /> 1`] = `
<label>
  <div
    class="iradio indeterminate"
  >
    <input
      style="opacity:0;position:absolute"
      type="radio"
      value=""
    />
    <ins
      class="iCheck-helper"
      role="presentation"
      style="position:absolute;top:0%;left:0%;display:block;width:100%;height:100%;margin:0;padding:0;background:#fff;border:0;opacity:0;cursor:pointer"
    />
  </div>
  <div>
    <span>
      label
    </span>
  </div>
</label>
`;
